exporter/sumologicexporter/exporter.go

```diff
@@ -178,3 +178,101 @@ func (se *sumologicexporter) pushLogsData(_ context.Context, ld pdata.Logs) (int
 
 	return 0, nil
 }
+
+func newMetricsExporter(
+	cfg *Config,
+	params component.ExporterCreateParams,
+) (component.MetricsExporter, error) {
+	se, err := initExporter(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return exporterhelper.NewMetricsExporter(
+		cfg,
+		params.Logger,
+		se.pushMetricsData,
+		// Disable exporterhelper Timeout, since we are using a custom mechanism
+		// within exporter itself
+		exporterhelper.WithTimeout(exporterhelper.TimeoutSettings{Timeout: 0}),
+		exporterhelper.WithRetry(cfg.RetrySettings),
+		exporterhelper.WithQueue(cfg.QueueSettings),
+	)
+}
+
+// pushMetricsData groups data with common metadata and send them as separate batched requests
+// it returns number of unsent metrics and error which contains list of dropped records
+// so they can be handle by the OTC retry mechanism
+func (se *sumologicexporter) pushMetricsData(_ context.Context, ld pdata.Metrics) (int, error) {
+	var (
+		errors         []error
+		sdr            = newSender(se.config, se.client, se.filter)
+		droppedRecords []metricPair
+		attributes     pdata.AttributeMap
+	)
+
+	// Iterate over ResourceMetrics
+	rms := ld.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		resource := rms.At(i)
+
+		if resource.IsNil() {
+			continue
+		}
+
+		attributes = resource.Resource().Attributes()
+
+		// iterate over InstrumentationLibraryMetrics
+		ilm := resource.InstrumentationLibraryMetrics()
```

Comment:
This is actually a slice, so frequently it's associated with `ilms` variable (not that it matters that much)